### PR TITLE
docs: update upstream network filters description

### LINKED ISSUE #11015: Clarify what network filters can be used as upstream filters
The docs for upstream filters imply that all network filters work: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/filter.proto#envoy-v3-api-msg-config-cluster-v3-filter

In reality, the network filter needs to be registered specifically as an upstream network filter: a `NamedUpstreamNetworkFilterConfigFactory` needs to be registered vs a `NamedNetworkFilterConfigFactory`. 

This should be made clearer in the docs.

### PR DESCRIPTION
**Pull request by rebrendov**
Signed-off-by: Tsimafei Bredau <timofei.bredov@gmail.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: update upstream network filters description to reflect availability of upstream filters
Additional Description: made documentation more clear on what upstream filters are currently available and removed a link to a list of network level filters, to avoid confusion
Risk Level: Low
Testing: ran a format checker with git push hook
Docs Changes: in description of upstream filters made more clear that no upstream filters are available in current version of envoy according to #11015 and #10525 
Release Notes: N/A
Fixes #11015

### CODE REVIEW

File: docs/root/intro/arch_overview/upstream/upstream_filters.rst [hunk 1]
```diff
@@ -4,8 +4,10 @@ Upstream network filters
 ========================
 
 Upstream clusters provide an ability to inject network level (L3/L4)
-:ref:`filters <arch_overview_network_filters>`. The filters apply to the
-connection to the upstream hosts, using the same API presented by listeners for
+filters. It should be noted that network fitler needs to
```
Comment (jmarantz) [inline docs/root/intro/arch_overview/upstream/upstream_filters.rst:7]: I was going to suggest some spelling and grammar tweaks, but I'm actually not sure about the semantics of this change.

I think I'll defer to @mattklein123 for this.

**Comment by jmarantz** [comment]
Also, please never force-push in Envoy github development. It makes reviewing very difficult.

Use normal push.

File: docs/root/intro/arch_overview/upstream/upstream_filters.rst (hunk 1 above)
Comment (rebrendov) [reply docs/root/intro/arch_overview/upstream/upstream_filters.rst:7]: > I was going to suggest some spelling and grammar tweaks
Yeah, I'm not sure what would be the best way to convey the information from #11015 in the docs.

I'm also not sure if the part about "registering upstream filters in the code" is necessary, but it felt strange to just add that there's no upstream filters without an explanation.

Comment (mattklein123) [reply docs/root/intro/arch_overview/upstream/upstream_filters.rst:7]: I think this is OK with me, modulo the spelling fixes needed (fitler). Thank you!

/wait

File: docs/root/intro/arch_overview/upstream/upstream_filters.rst [hunk 2]
```diff
@@ -4,8 +4,10 @@ Upstream network filters
 ========================
 
 Upstream clusters provide an ability to inject network level (L3/L4)
-:ref:`filters <arch_overview_network_filters>`. The filters apply to the
-connection to the upstream hosts, using the same API presented by listeners for
+filters. It should be noted that network filter needs to
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/upstream/upstream_filters.rst:7]:
```suggestion
filters. It should be noted that a network filter needs to
```

Comment (rebrendov) [reply docs/root/intro/arch_overview/upstream/upstream_filters.rst]: Fixed. Ran this text through a grammar checker, just to make sure that I didn't miss anything else.

**Pull request merged**